dpctl/_sycl_device.pyx

```diff
@@ -706,7 +706,12 @@ cdef class SyclDevice(_SyclDevice):
             the sub-devices.
         """
         cdef DPCTLDeviceVectorRef DVRef = NULL
-        DVRef = DPCTLDevice_CreateSubDevicesEqually(self._device_ref, count)
+        if count > 0:
+            DVRef = DPCTLDevice_CreateSubDevicesEqually(self._device_ref, count)
+        else:
+            raise ValueError(
+                "Creating sub-devices with zero compute units is requested"
```

Comment:
What will Cython do here? Will it raise the exception of silently cause havoc?

Reply:
the `cdef list func_name(...)` will correctly propagate the Python error raised.